src/shared_realm.hpp

```diff
@@ -158,10 +164,12 @@ class Realm : public std::enable_shared_from_this<Realm> {
         // Optional schema for the file.
         // If the schema and schema version are supplied, update_schema() is
         // called with the supplied schema, version and migration function when
-        // the Realm is actually opened and not just retreived from the cache
+        // the Realm is actually opened and not just retrieved from the cache
         util::Optional<Schema> schema;
         uint64_t schema_version = -1;
         MigrationFunction migration_function;
+        // Called when the schema changes are delivered to this Realm instance.
+        SchemaChangedFunction schema_changed_function;
```

Comment:
After thinking about how we'd use this in Cocoa a bit, it occurred to me that we actually need to notify both the binding and the end user to properly support a truly dynamic schema, since the Browser also needs to know when the schema has changed. I think this strengthens my view that notifying the binding should be done via the context.